src/functions.js

```diff
@@ -12,45 +18,117 @@ function getNumberOfBikes( document ) {
+
+    } // and return as an array of strings
+ return buttonList;
+ 
   // https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection
   // also you may want to convert the HTMLCollection to an array
+  
 };
 
 function getNavLinksText( document ){
+  var navBar= document.getElementById('navbarSupportedContent');
+  var links= navBar.getElementsByTagName('a');
+  var newArray = [];
+
+  for (i=0; i<links.length; i++){
+    newArray.push(links[i].textContent.trim());
+
+  }
+return newArray;
   // get the text content of all 'A' tags inside element with id 'navbarSupportedContent'
   // reuturn as an array of strings
 }
 
 function addDashesToLinks( document ){
   // surround the text in navigation bar links with '-'
   // no return needed
+  var links = document.querySelectorAll('.nav-link');
+  links.forEach(function(item){
+   
+    item.textContent ='-' + item.textContent + '-';
+  });
+ 
 }
 
 function italicTitles( document ){
-  // convert links in 'Upcoming Events' section to italic using `<i>` tag
-  // no return needed
-}
+
+//console.log(document.all)
+  var title =document.querySelectorAll(".article-title--sidebar a")
+  title.forEach(function(item){
+     item.innerHTML='<i>'+ item.innerHTML +'</i>';
+   })
+
+  
+
+  }
 
 function greenLinks( document ){
+  var elements = document.querySelectorAll(".article-read-more a");
+  elements.forEach(function(item) {
+  item.style.color = "green";
+});
   // make `Learn more` links green
   // no return needed
 }
 
 function addLink( document ){
+
+  var newulTag = document.createElement('ul'); //create ul tag
+  newulTag.setAttribute("class", "navbar-nav mr-auto")
+
+  var newLiTag = document.createElement('li'); //create li tag
+  newLiTag.setAttribute("class", "nav-item")
+  newulTag.appendChild(newLiTag)
+  
+  var newATag= document.createElement('a');// create an A tag
+  newATag.setAttribute('class', 'nav-link');
+  newATag.setAttribute('href','https://codeyourfuture.io/'); 
+   
+  var text = document.createTextNode ('Code Your Future');
+  newATag.appendChild(text)
+  newLiTag.appendChild(newATag)
+  newulTag.appendChild(newLiTag);  //<a> added into the <li> element
+
+   //create new class
+
+                  
+  
+
+
+var navList= document.getElementsByClassName('navbar-nav mr-auto')[0];
+navList.appendChild(newulTag);
+
+
+
+
   // Using `createElement` etc. create a new navbar item link 'Code Your Future' which links to
   // `https://codeyourfuture.io/`. It should have same structure as the other links
   // no return needed
 }
 
-module.exports = {
-  getTitle,
-  getNumberOfBikes,
-  getAllButtonText,
-  getNavLinksText,
-  addDashesToLinks,
-  italicTitles,
-  greenLinks,
-  addLink
+if (typeof module !== 'undefined') {
+  module.exports = {
+    getTitle,
+    getNumberOfBikes,
+    getAllButtonText,
+    getNavLinksText,
+    addDashesToLinks,
+    italicTitles,
+    greenLinks,
+    addLink
+  }
 };
+//addLink( document )
```

Comment:
remove the comments here and the function calls you created to make sure it was working.
Extra challenge: link up the function calls to new buttons on your page, so that you can execute the functions when you click them!

Reply:
@robota-x it doesn't work without the function call